src/core/meshcop/commissioner.cpp

```diff
@@ -1173,8 +1292,25 @@ const char *Commissioner::StateToString(otCommissionerState aState)
 
 void Commissioner::LogJoinerEntry(const char *aAction, const Joiner &aJoiner) const
 {
-    otLogInfoMeshCoP("%s Joiner (%s, %s)", aAction, aJoiner.mAny ? "*" : aJoiner.mEui64.ToString().AsCString(),
-                     aJoiner.mPsk);
+    switch (aJoiner.mType)
+    {
+    case Joiner::kTypeUnused:
```

Comment:
We can, but since this method is only for logging/debugging I think it's ok to not assert to avoid code size increase and also avoid the messiness of assert in switch requiring `OT_UNREACHABLE_CODE(break)` (to make it work for some toolchains)

Reply:
Makes sense to me.